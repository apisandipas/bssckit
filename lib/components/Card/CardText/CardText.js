"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledTools = require("styled-tools");

var _theme = _interopRequireDefault(require("../../../theme"));

var _base = require("../../Utilities/base");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  margin-top: ", ";\n  margin-bottom: ", ";\n  &:last-child {\n    margin-bottom: ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var CardText = (0, _styledComponents["default"])(_base.P)(_templateObject(), (0, _styledTools.theme)('cardText.margin.top'), (0, _styledTools.theme)('cardText.margin.bottom'), (0, _styledTools.theme)('cardText.margin.lastChildBottom'));
CardText.defaultProps = {
  theme: _theme["default"]
};
var _default = CardText;
exports["default"] = _default;