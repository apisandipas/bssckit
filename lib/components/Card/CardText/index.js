"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "CardText", {
  enumerable: true,
  get: function get() {
    return _CardText["default"];
  }
});

var _CardText = _interopRequireDefault(require("./CardText"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }