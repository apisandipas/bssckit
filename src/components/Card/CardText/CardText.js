import styled from 'styled-components';
import { theme } from 'styled-tools';
import defaultTheme from '../../../theme';
import { P } from '../../Utilities/base';

const CardText = styled(P)`
  margin-top: ${theme('cardText.margin.top')};
  margin-bottom: ${theme('cardText.margin.bottom')};
  &:last-child {
    margin-bottom: ${theme('cardText.margin.lastChildBottom')};
  }
`;

CardText.defaultProps = {
  theme: defaultTheme,
};

export default CardText;
