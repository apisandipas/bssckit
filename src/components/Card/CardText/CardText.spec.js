import React from 'react';
import { axe } from 'jest-axe';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import CardText from './CardText';

describe('CardText Component', () => {
  it('matches CardText', async () => {
    const { container } = render(<CardText />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('margin-top', '0');
    expect(container.firstChild).toHaveStyleRule('margin-bottom', '1rem');
    expect(container.firstChild).toHaveStyleRule('margin-bottom', '0', {
      modifier: '&:last-child',
    });
  });
});
