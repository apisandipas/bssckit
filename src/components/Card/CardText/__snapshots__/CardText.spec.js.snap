// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardText Component matches CardText 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 1rem;
}

.c0:last-child {
  margin-bottom: 0;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<p
  class="sc-AxjAm c0"
/>
`;
